python/lsst/daf/butler/registry/sql_registry.py

```diff
@@ -276,7 +276,7 @@ def isWriteable(self) -> bool:
 
     def copy(self, defaults: RegistryDefaults | None = None) -> SqlRegistry:
         """Create a new `SqlRegistry` backed by the same data repository
-        and connection as this one, but independent defaults.
+        as this one, but independent defaults and database connection.
```

Comment:
Yeah, what you described matches my understanding.  `sqlalchemy.Engine` guarantees that only one caller has a given "logical" connection checked out at a time, though you're correct that it doesn't necessarily map to a separate network connection.  

To be clear, the part that is now independent is that each Database instance will use a different `sqlalchemy.engine.Connection` object, whereas before they would sometimes be accidentally shared via the `Database._session_connection` instance variable.  This was problematic for a few reasons:
1. `Connection` is specifically documented to not be threadsafe
2. The transaction state would get mixed together between the threads because this is managed via the `Connection` object
3. Depending on the timing, one thread could close the `Connection` object before the other thread was done using it, or replace it with a different connection while another thread was in the middle of using it.

Perhaps a better phrasing would be "independent database session"... I'll tweak the comment.